apps/rocketjoed/main.cpp

```diff
@@ -193,13 +194,29 @@ int main(int argc, char* argv[]) {
         process_pool.add_worker_process(command_line["worker_number"].as<std::size_t>()); /// todo hack
     }
 
-    boost::asio::signal_set sigint_set(env.loop(), SIGINT, SIGTERM);
-    sigint_set.async_wait(
-        [&sigint_set](const boost::system::error_code& /*err*/, int /*num*/) {
-            sigint_set.cancel();
-        });
-
-    env.startup();
+    switch (cfg_.python_configuration_.mode_) {
+        case components::sandbox_mode_t::none: {
+            log.error("else if( cfg_.python_configuration_.mode_ == components::sandbox_mode_t::none)");
+            return 1;
+        }
+        case components::sandbox_mode_t::script: {
+            components::python_interpreter vm(cfg_.python_configuration_,log);
+            vm.init();
+            vm.run_script(all_args);
+            break;
+        }
+        case components::sandbox_mode_t::jupyter_kernel:
+        case components::sandbox_mode_t::jupyter_engine: {
+            boost::asio::signal_set sigint_set(env.loop(), SIGINT, SIGTERM);
+            sigint_set.async_wait(
+                [&sigint_set](const boost::system::error_code& /*err*/, int /*num*/) {
+                    sigint_set.cancel();
+                });
+
+            env.startup();
+            break;
+        }
+    }
```

Comment:
Maybe it’s better to add default behavior so that when adding a new item to the enumeration, the code doesn't break?

Reply:
It's a good idea to add a default+ break.